coil-core/src/commonMain/kotlin/coil3/Uri.kt

```diff
@@ -32,6 +32,49 @@ class Uri internal constructor(
     }
 }
 
+/**
+ * A builder for creating or modifying [Uri] instances.
+ * This provides a more convenient way to manipulate URI components.
+ */
+class UriBuilder internal constructor(
+    private var scheme: String?,
+    private var authority: String?,
+    private var path: String?,
+    private var queryParameters: MutableMap<String, String>,
+    private var fragment: String?
+) {
+    fun scheme(scheme: String?): UriBuilder = apply { this.scheme = scheme }
+    fun authority(authority: String?): UriBuilder = apply { this.authority = authority }
+    fun path(path: String?): UriBuilder = apply { this.path = path }
+    fun fragment(fragment: String?): UriBuilder = apply { this.fragment = fragment }
+
+    fun appendQueryParameter(key: String, value: String): UriBuilder = apply {
```

Comment:
Let's remove `appendQueryParameter` and `clearQuery` for now and only expose a `query` method - similar to the other methods. I think we might have to handle encoding/decoding when adding query params similar to Android's Uri class, but I'm not 100% sure. Let's offer basic `Uri` manipulation methods for now and we can add other methods later.